src/cljs/ataru/hakija/application_handlers.cljs

```diff
@@ -1120,3 +1125,21 @@
   :application/setup-window-unload
   (fn [_ _]
     {:set-window-close-callback nil}))
+
+(reg-event-fx
+  :application/update-hakuaika-left
+  (fn [{db :db} _]
+    (let [hakuaika-end             (->> db :form :tarjonta :hakukohteet
+                                        (map :hakuaika)
+                                        (filter :on)
+                                        (sort-by :end >)
+                                        first
+                                        :end)
+          hakuaika-left-in-seconds (c/in-seconds (c/interval (c/now) (from-long hakuaika-end)))
+          relevant-hakuaika-left   (when (and (> (* 24 3600) hakuaika-left-in-seconds)
+                                              (< 0 hakuaika-left-in-seconds))
+                                     hakuaika-left-in-seconds)]
+      {:db             (assoc-in db [:form :hakuaika-left] relevant-hakuaika-left)
```

Comment:
Jäin miettimään onnistuisko tän tekeminen niin, että app-db:n tila ei muuttuisi automaattisesti sekunnin välein, vaan päivittyvä aika ois esim. komponentin sisäisessä tilassa jonkin sortin reactionissa. En kyllä suoriltaan keksinyt mitään supereleganttia tapaa tähän.